var/spack/repos/builtin/packages/raja/package.py

```diff
@@ -37,8 +37,8 @@ class Raja(CMakePackage, CudaPackage, ROCmPackage):
     # and remove the +tests conflict below.
     variant('tests', default=False, description='Build tests')
 
-    depends_on('blt@0.4.0:', type='build', when='@0.13.1:')
-    depends_on('blt@:0.3.6', type='build', when='@:0.13.0')
+    depends_on('blt@0.4.0:', type='build', when='@0.13.0:')
+    depends_on('blt@:0.3.6', type='build', when='@:0.12.0')
```

Comment:
this removed any depends_on for 0.12.1

Reply:
I wanted to allow building `@0.12.1` with `blt@0.4.0`, but I didn't want to force building with it (i.e. I wanted to allow building `0.12.1` with earlier `blt` versions if desired). Does that seem reasonable to you?

Reply:
I understand I am just saying that you are missing raja 0.12.1 from your when statements.  I think this is currently working for you because something else depends on BLT and so its in the spec somewhere.

It may be defaulting to the submodule BLT, I'm not completely sure what comes out of `spec['blt`]` if you don't say it depends on BLT.

As it is, if you built RAJA by itself, RAJA would not depend on BLT at all.

Reply:
> I understand I am just saying that you are missing raja 0.12.1 from your when statements. 

Whoops! you are definitely right - I'll fix that

Reply:
I have added a "catch-all" dependency on `blt` so this should be addressed now